rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/AddClassFileMetadata.java

```diff
@@ -81,12 +82,12 @@ public void perform(GraphRewrite event, EvaluationContext context, JavaClassFile
                     for (final String interfaceName : interfaceNames)
                     {
                         JavaClassModel interfaceModel = javaClassService.getOrCreatePhantom(interfaceName);
-                        javaClassModel.addImplements(interfaceModel);
+                        javaClassService.addInterface(javaClassModel, interfaceModel);
                     }
                 }
 
                 String superclassName = bcelJavaClass.getSuperclassName();
-                if (!StringUtils.isBlank(superclassName))
+                if (!bcelJavaClass.isInterface() && !StringUtils.isBlank(superclassName))
```

Comment:
What happened if the interface was let in? Is the superInterface registered now?